File: mobile/lib/dtos/payment_intent_response.dart
class PaymentIntentResponse {
  String paymentIntentId;
  String clientSecret;
  int userId;
  int amount;
  String currency;
  DateTime createdAt;

  PaymentIntentResponse({
    required this.paymentIntentId,
    required this.clientSecret,
    required this.userId,
    required this.amount,
    required this.currency,
    required this.createdAt,
  });

  factory PaymentIntentResponse.fromJson(Map<String, dynamic> json) =>
      PaymentIntentResponse(
        paymentIntentId: json["paymentIntentId"],
        clientSecret: json["clientSecret"],
        userId: json["userId"],
        amount: json["amount"],
        currency: json["currency"],
        createdAt: DateTime.parse(json["createdAt"]),
      );

  Map<String, dynamic> toJson() => {
        "paymentIntentId": paymentIntentId,
        "clientSecret": clientSecret,
        "userId": userId,
        "amount": amount,
        "currency": currency,
        "createdAt": createdAt.toIso8601String(),
      };
}

File: backend/MemoWise.Model/DTO/PaymentIntentResponse.cs
namespace MemoWise.Model.DTO;

public class PaymentIntentResponse
{
    public string PaymentIntentId { get; set; }
    public string ClientSecret { get; set; }
    public int UserId { get; set; }
    public long Amount { get; set; }
    public string Currency { get; set; }
    public DateTime CreatedAt { get; set; }
}
